packages/user-interface/src/pages/MessagePage.tsx

```diff
@@ -1,59 +1,77 @@
-import { useGetTakenV2Query, TaakV2 } from "@nl-portal/nl-portal-api";
+import {
+  Bericht,
+  TaakV2,
+  useGetBerichtQuery,
+  useGetTakenV2Query,
+} from "@nl-portal/nl-portal-api";
 import { Paragraph } from "@gemeente-denhaag/typography";
 import { FormattedMessage } from "react-intl";
 import { useOutletContext, useParams } from "react-router-dom";
 import BackLink from "../components/BackLink";
 import PageGrid from "../components/PageGrid";
 import PageHeader from "../components/PageHeader";
 import TasksList from "../components/TasksList";
-import { MessageType } from "../components/Message";
-import { messages } from "./MessagesPage";
 import { RouterOutletContext } from "../contexts/RouterOutletContext";
+import styles from "./MessagePage.module.scss";
+import Skeleton from "../components/Skeleton.tsx";
 
 const MessagePage = () => {
   const { id } = useParams();
   const { paths } = useOutletContext<RouterOutletContext>();
-  const {
-    data: tasksData,
-    loading: taskLoading,
-    error: tasksError,
-  } = useGetTakenV2Query({
+  const { data: tasksData, loading: taskLoading } = useGetTakenV2Query({
     variables: { zaakId: id },
   });
-  const loading = taskLoading;
-  const error = tasksError;
-  const message: MessageType = messages.find((m) => m.id === Number(id))!;
-  const tasks = tasksData?.getTakenV2.content as TaakV2[] | undefined;
+  const {
+    data: messageData,
+    loading: messageLoading,
+    error: messageError,
+  } = useGetBerichtQuery({
+    variables: { id: id },
+  });
+  const tasks: TaakV2[] | undefined = tasksData?.getTakenV2.content as
+    | TaakV2[]
+    | undefined;
+  const message: Bericht | undefined = messageData?.getBericht as
+    | Bericht
+    | undefined;
+
+  if (messageLoading) {
```

Comment:
Please use the loading and error states from the PageHeader and Taskslist.

Reply:
Should I remove the tasklist from the page as a whole? They are not part of a bericht anyway.

Reply:
Yes! They are not used right now (for the hackathon). We can add them later.

Reply:
> Please use the loading and error states from the PageHeader
Can you elaborate on this? there are no states from the pageHeader in this component right now.

Reply:
PageHeader has a loading state. For error the message will be empty so no "onderwerp" so no header :) 

I think the error fits best inside your new MessageDetail component.

Reply:
but for message fetching error, we don't even render the PageHeader, we only render the error message with a back link. so I'm not understanding something here...
There is no point in rendering the header if the message doesn't resolve.

Reply:
In my opinion the loading and error state shouldn't be on page lvl. But should be part of the components on the page itself. So please remove them here and add them to your new message details component.

Reply:
Thanks, that makes so much more sense. Sorry for not understanding sooner.